src/onnx/parse_softmaxcrossentropyloss.cpp

```diff
@@ -0,0 +1,316 @@
+
+reduction : string (default is mean)
+                Type of reduction to apply to loss: none, sum, mean(default).
+                - 'none': no reduction will be applied
+                - 'sum': the output will be summed.
+                - 'mean': the sum of the output will be divided by
+                        the number of elements in the output.
+
+Inputs (2 - 3)
+++++++++++++++
+
+scores (differentiable) : T
+    The predicted outputs with shape [batch_size, class_size], or [batch_size, class_size, D1, D2 ,
+..., Dk], where K is the number of dimensions.
+
+labels (non-differentiable) : Tind
+    The ground truth output tensor, with shape [batch_size], or [batch_size, D1, D2, ..., Dk],
+    where K is the number of dimensions. Labels element value shall be in range of [0, C).
+    If ignore_index is specified, it may have a value outside [0, C) and the label values should
+    either be in the range [0, C) or have the value ignore_index.
+
+weights (optional, non-differentiable) : T
+    A manual rescaling weight given to each class. If given,
+    it has to be a 1D Tensor assigning weight to each of the classes.
+    Otherwise, it is treated as if having all ones.
+
+Outputs (1 - 2)
+==================
+
+output (differentiable) : T
+    Weighted loss float Tensor. If reduction is 'none', this has the shape of [batch_size],
+    or [batch_size, D1, D2, ..., Dk] in case of K-dimensional loss. Otherwise, it is a scalar.
+
+log_prob (optional, differentiable) : T
+    Log probability tensor. If the output of softmax is prob, its value is log(prob).
+
+Type Constraints
+===================
+    T : tensor(float16), tensor(float), tensor(double), tensor(bfloat16) (Currently not supported in
+MIGX) Constrain input and output types to float tensors.
+
+Tind : tensor(int32), tensor(int64)
+    Constrain target to integer types
+*/
+
+#include <migraphx/onnx/op_parser.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/onnx/checks.hpp>
+#include <migraphx/instruction.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+struct parse_softmaxcrossentropyloss : op_parser<parse_softmaxcrossentropyloss>
+{
+    std::vector<op_desc> operators() const { return {{"SoftmaxCrossEntropyLoss"}}; }
+
+    std::vector<instruction_ref> parse(const op_desc& /*opd */,
+                                       const onnx_parser& parser,
+                                       const onnx_parser::node_info& info,
+                                       const std::vector<instruction_ref>& args) const
+    {
+        // Get and handle attributes
+        std::string reduction = "mean";
+        if(contains(info.attributes, "reduction"))
+        {
+            std::set<std::string> supported_modes = {"mean", "none", "sum"};
+
+            reduction = info.attributes.at("reduction").s();
+            if(not contains(supported_modes, reduction))
```

Comment:
Just put the support modes inline: `if(not contains({"mean", "none", "sum"}, reduction))`

Reply:
Done